lib/services/project-data-service.ts

```diff
@@ -51,6 +53,137 @@ export class ProjectDataService implements IProjectDataService {
+		const iOSAssetStructure = await this.getIOSAssetsStructure(opts);
+		const androidAssetStructure = await this.getAndroidAssetsStructure(opts);
+
+		this.$logger.trace("iOS Assets structure:", JSON.stringify(iOSAssetStructure, null, 2));
+		this.$logger.trace("Android Assets structure:", JSON.stringify(androidAssetStructure, null, 2));
+
+		return {
+			ios: iOSAssetStructure,
+			android: androidAssetStructure
+		};
+	}
+
+	@exported("projectDataService")
+	public async getIOSAssetsStructure(opts: IProjectDir): Promise<IAssetGroup> {
+		const projectDir = opts.projectDir;
+		const projectData = this.getProjectData(projectDir);
+
+		const basePath = path.join(projectData.appResourcesDirectoryPath, this.$devicePlatformsConstants.iOS, AssetConstants.iOSAssetsDirName);
+		const pathToIcons = path.join(basePath, AssetConstants.iOSIconsDirName);
+		const icons = await this.getIOSAssetSubGroup(pathToIcons);
+
+		const pathToSplashBackgrounds = path.join(basePath, AssetConstants.iOSSplashBackgroundsDirName);
+		const splashBackgrounds = await this.getIOSAssetSubGroup(pathToSplashBackgrounds);
+
+		const pathToSplashCenterImages = path.join(basePath, AssetConstants.iOSSplashCenterImagesDirName);
+		const splashCenterImages = await this.getIOSAssetSubGroup(pathToSplashCenterImages);
+
+		const pathToSplashImages = path.join(basePath, AssetConstants.iOSSplashImagesDirName);
+		const splashImages = await this.getIOSAssetSubGroup(pathToSplashImages);
+
+		return {
+			icons,
+			splashBackgrounds,
+			splashCenterImages,
+			splashImages
+		};
+	}
+
+	@exported("projectDataService")
+	public async getAndroidAssetsStructure(opts: IProjectDir): Promise<IAssetGroup> {
+		// TODO: Use image-size package to get the width and height of an image.
+		// TODO: Parse the splash_screen.xml in nodpi directory and get from it the names of the background and center image.
+		// TODO: Parse the AndroidManifest.xml to get the name of the icon.
+		// This way we'll not use the image-definitions.json and the method will return the real android structure.
+		const projectDir = opts.projectDir;
+		const projectData = this.getProjectData(projectDir);
+		const pathToAndroidDir = path.join(projectData.appResourcesDirectoryPath, this.$devicePlatformsConstants.Android);
+		const hasMigrated = this.$androidResourcesMigrationService.hasMigrated(projectData.appResourcesDirectoryPath);
+		const basePath = hasMigrated ? path.join(pathToAndroidDir, SRC_DIR, MAIN_DIR, RESOURCES_DIR) : pathToAndroidDir;
+
+		const currentStructure = this.$fs.enumerateFilesInDirectorySync(basePath);
+		const content = this.getImageDefinitions().android;
+
+		return {
+			icons: this.getAndroidAssetSubGroup(content.icons, currentStructure),
+			splashBackgrounds: this.getAndroidAssetSubGroup(content.splashBackgrounds, currentStructure),
+			splashCenterImages: this.getAndroidAssetSubGroup(content.splashCenterImages, currentStructure),
+			splashImages: null
+		};
+	}
+
+	private getImageDefinitions(): any {
```

Comment:
I believe this method returns `IAssetsStructure`

Reply:
unfortunately it does not :(